provider_packages/setup_provider_packages.py

```diff
@@ -788,21 +808,39 @@ def convert_cross_package_dependencies_to_table(
     "release_version release_version_no_leading_zeros last_commit_hash content file_name")
 
 
-def strip_leading_zeros(release_version: str) -> str:
-    return release_version.replace(".0", ".")
+def strip_leading_zeros(release_version: str, backport_packages: bool) -> str:
```

Comment:
This doesn't look like it replaces leading zeros, but all zeros.

Reply:
Not really :). Only 0's after actual ".". It's only used in calver and I used it to replace 2020.04.03 to 2020.4.3 because (I think) PIP did not work correctly with "0" in version :(. But this is only for calver. I will change it slightly though to be claearer.

Reply:
This still works for dates beacause there are no dates with .0 :)

Reply:
```
def strip_leading_zeros_in_calver(calver_version: str) -> str:
    """
    Strips leading zeros from calver version number.

    This converts 1974.04.03 to 1974.4.3 as the format with leading month and day zeros is not accepted
    by PIP versioning.

    :param calver_version: version number in calver format (potentially with leading 0s in date and month)
    :return: string with leading 0s after dot replaced.
    """
    return calver_version.replace(".0", ".")
```